.github/workflows/ci.yml

```diff
@@ -51,10 +51,15 @@ jobs:
       run: |
         mkdir build
         cd build
+        # Only build with sanitizers on Debug
+        if [[ "${{ matrix.build-type }}" == "Debug" ]]; then
```

Comment:
I think this is fine for now, but in general I've found that it's easy to lose track of these sorts of hidden matrix expansions inside of GitHub Actions.

Two ideas:

1. Refactor the sanitizer setup to make `PEPARSE_USE_SANITIZER` into a bool setting and provide a separate variable that contains the actual list of sanitizers to run (like `PEPARSE_ENABLED_SANITIZERS=Address,Undefined`).

2. Refactor this to use a separate step, something like:

    ```yaml
    - name: env
      if: matrix.build-type == "Debug"
      run: |
        echo "SAN_FLAG=Address,Undefined" >> $GITHUB_ENV
    ```
    
    (based on these docs: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#environment-files)

But like I said, fine for now. Just something to consider for the future.

Reply:
I'll make something like these changes in this PR, since I think it'll be easier to understand for anyone who gets errors only with "Debug" matrix (due to sanitizers) 👍